libyul/backends/wasm/WordSizeTransform.cpp

```diff
@@ -0,0 +1,275 @@
+
+#include <libdevcore/CommonData.h>
+
+#include <array>
+
+using namespace std;
+using namespace dev;
+using namespace yul;
+
+void WordSizeTransform::operator()(FunctionDefinition& _fd)
+{
+	rewriteTypedNameList(_fd.parameters);
+	rewriteTypedNameList(_fd.returnVariables);
+	(*this)(_fd.body);
+}
+
+void WordSizeTransform::operator()(FunctionalInstruction& _ins)
+{
+	rewriteFunctionCallArguments(_ins.arguments);
+}
+
+void WordSizeTransform::operator()(FunctionCall& _fc)
+{
+	rewriteFunctionCallArguments(_fc.arguments);
+}
+
+void WordSizeTransform::operator()(If&)
+{
+	yulAssert(false, "If statement not implemented.");
+}
+
+void WordSizeTransform::operator()(Switch&)
+{
+	yulAssert(false, "Switch statement not implemented.");
+}
+
+void WordSizeTransform::operator()(Block& _block)
+{
+	iterateReplacing(
+		_block.statements,
+		[&](Statement& _s) -> boost::optional<vector<Statement>>
+		{
+			if (_s.type() == typeid(VariableDeclaration))
+			{
+				VariableDeclaration& varDecl = boost::get<VariableDeclaration>(_s);
+				if (
+					!varDecl.value ||
+					varDecl.value->type() == typeid(FunctionalInstruction) ||
+					varDecl.value->type() == typeid(FunctionCall)
+				)
+				{
+					if (varDecl.value) visit(*varDecl.value);
+					rewriteTypedNameList(varDecl.variables);
+					return boost::none;
+				}
+				else if (varDecl.value->type() == typeid(Identifier))
+				{
+					yulAssert(varDecl.variables.size() == 1, "");
+					auto newRhs = expandIdentifierValue(boost::get<Identifier>(*varDecl.value));
+					auto newLhs = generateNewNames(varDecl.variables[0].name);
+					vector<Statement> ret;
+					for (int i = 0; i < 4; i++)
+						ret.push_back(
+							VariableDeclaration{
+								varDecl.location,
+								{TypedName{varDecl.location, newLhs[i], "u64"_yulstring}},
+								std::move(newRhs[i])
+							}
+						);
+					return ret;
+				}
+				else if (varDecl.value->type() == typeid(Literal))
+				{
+					yulAssert(varDecl.variables.size() == 1, "");
+					auto newRhs = expandLiteralValue(boost::get<Literal>(*varDecl.value));
+					auto newLhs = generateNewNames(varDecl.variables[0].name);
+					vector<Statement> ret;
+					for (int i = 0; i < 4; i++)
+						ret.push_back(
+							VariableDeclaration{
+								varDecl.location,
+								{TypedName{varDecl.location, newLhs[i], "u64"_yulstring}},
+								std::move(newRhs[i])
+							}
+						);
+					return ret;
+				}
+			}
+			else if (_s.type() == typeid(Assignment))
+			{
+				Assignment& assignment = boost::get<Assignment>(_s);
+				yulAssert(assignment.value, "");
+				if (
+					assignment.value->type() == typeid(FunctionalInstruction) ||
+					assignment.value->type() == typeid(FunctionCall)
+				)
+				{
+					if (assignment.value) visit(*assignment.value);
+					rewriteIdentifierList(assignment.variableNames);
+					return boost::none;
+				}
+				else if (assignment.value->type() == typeid(Identifier))
+				{
+					yulAssert(assignment.variableNames.size() == 1, "");
```

Comment:
Is `expandIdentifierValue` / `expandLiteralValue` the only difference between this and the next block? If yes, then please combine the two.